### PR TITLE
chore: use .clang-format

### PR DESCRIPTION
**Pull request by taegyunkim**
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,245 @@
+---
```
Comment (sanchda) [inline .clang-format:1]: Is this just a dump of the format that was previously hardcoded in the helper script?

Comment (taegyunkim) [reply .clang-format:1]: Yes, generated by `clang-format --style="{BasedOnStyle: Mozilla, IndentWidth: 4, ColumnLimit: 120}" --dump-config > .clang-format` using the version 18.1.5 as specified in [hatch.toml](https://github.com/DataDog/dd-trace-py/blob/2f7063bc3cb3ddacbc97c0e53627944d63bcf24b/hatch.toml#L27)

**Comment by erikayasuda** [comment]
> `CODEOWNERS` have been resolved as:
> 
> ```
> .clang-format                                                           @DataDog/apm-core-python
> hooks/scripts/run-clang-format.sh                                       @DataDog/apm-core-python
> scripts/cformat.sh                                                      @DataDog/apm-core-python
> ```

Should we mark `.clang-format` to be owned by the guild?

**Review by brettlangdon — APPROVED** [review]

**Comment by brettlangdon** [comment]
> > `CODEOWNERS` have been resolved as:
> > ```
> > .clang-format                                                           @DataDog/apm-core-python
> > hooks/scripts/run-clang-format.sh                                       @DataDog/apm-core-python
> > scripts/cformat.sh                                                      @DataDog/apm-core-python
> > ```
> 
> Should we mark all of these clang files to be owned by the guild?

That makes sense to me....

**Pull request merged**